### ISSUE TITLE
Investigate and Benchmark Aspera vs HTTP Downloads

### ISSUE THREAD

**Issue by Miserlou** [data-refinery/data-refinery#91]
The European Nucleotide Archives offers 'Aspera' as an alternative to HTTP/FTP downloads. Properly optimized, downloads with Aspera could potentially be 82000x (?!) faster than TCP/HTTP - so, I think we should investigate and compare these use cases. Aspera uses dynamic bandwith control over SSH in combination multiplexed UDP streams to achieve maximum bandwidth usage with minimal overhead usage versus normal monoplexed HTTP, which wasn't designed for our kind of large file transfer.

The Aspera command line client, `ascp`, is available for download here: http://download.asperasoft.com/download/sw/cli/3.7.7/aspera-cli-3.7.7.608.927cce8-linux-64-release.sh

A download example looks like this:

```bash
$ ascp -T -l 300m -P33001 -i /etc/aperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:/vol1/ERA012/ERA012008/sff/library08_GJ6U61T06.sff .
```

And a bandwidth test looks like this:

```bash
$ ascp -wf -P 22 -i/etc/aperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:/
```

It'd be great if we could use `bench` or similar to compare this with an equivalent `curl`, `wget` or python `requests` download from ENA to an EC2 instance.

Unfortunately, the `ascp` client is proprietary software, so we wouldn't be able to redistribute a public image containing the installed software until we investigated the copyright implications of that.

**Comment by Miserlou** [comment]
Aspera downloads have been merged in via #109!

The results of my benchmarks are pretty staggering. Downloading a test file via FTP and Aspera on a t2.micro in us-east-2c:

Protocol         | Average Speed           | Download Time     |
--------------------|------------------|-----------------------|
FTP | 57.3KB/s   | 3h17m0s   |
Aspera (-T) [with encryption] | 363Mb/s   | 0h0m13s   |
Aspera (+T) [without encryption] | 466Mb/s   | 0h0m10s   |

Results were the same for downloads from ENA and NCBI, ENA actually being slightly faster (145M/s for NCBI, in part due to their encryption requirement which ENA doesn't have).

I also found that multiple Aspera downloads across multiple (n=2) separate instances did double the total speed to over 1Gb/s. Running multiple simultaneous Aspera downloads from a single machine did not significantly improve download speeds and seemed to max out at around 500Mb/s, as expected.

I also experimented with difference instance types, and found that the `m4.large` instance actually performed 100Mb/s _worse_ than the download on the `t2.micro` instances, despite having a "High" versus "Low" network bandwidth allowance, according to Amazon. I believe that this may because the `t2` instances were in `us-east-2c` and the `m4` was in `us-east-2b`, so this may be something to consider in future.

**tl;dr:** Where possible, downloaders have been switched to use Aspera, which is resulting in downloads that are **1182x** faster than FTP, thereby bringing our project back to the realm of feasibility. This feature has already been merged into mainline.
